src/main/java/com/dataguise/saas/service/ConnectionService.java

```diff
@@ -0,0 +1,450 @@
+                        if (currDbConnection.getPassword()
+                            .equals(updatedConnection.getPassword())) {
+                            // set password to "" if the password isn't changed.
+                            // Don't send encrypted password from saas ui to re encrypt.
+                            updatedConnection.setPassword("");
+                        }
+                    }
+                }
+                updatedConnection.setModule(connectionDTO.getConnection().getModule());
+                connectionId = restDriver.editConnection(connectionId, updatedConnection);
+            }
+        } catch (DgDrvException e) {
+            logger.error("Edit connection failed: " + e);
+            throw new BadGatewayException(e.getMessage(), HttpStatus.BAD_GATEWAY);
+        }
+        logger.info("Connection edited with id: " + connectionId);
+        try {
+            if (defaultSchedule != null && defaultSchedule == true) {
+                getScheduleInfo(connectionId);
+                SchedulerDTO schedule = getScheduleInfo(connectionId);
+                logger.info("Creating a schedule for connection with id: " + connectionId);
+                Boolean scheduleStatus = createDefaultDetectionSchedule(schedule,
+                    connectionDTO.getConnection().getModule());
+                if (scheduleStatus == true) {
+                    logger.info("Default Schedule task for connection id " + connectionId
+                        + " completed successfully.");
+                } else {
+                    logger.error(
+                        "Default schedule task for connection id " + connectionId + " failed.");
+                }
+            } else {
+                logger.info("Default Schedule is null");
+            }
+        } catch (Exception e) {
+            logger.error("Connection edited successfully but failed to schedule a task. " + e);
+            throw new InternalServerException(
+                ErrorConstants.ERROR_UPDATED_CONNECTION_WITH_SCHEDULER.getErrorMessage(),
+                HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+        return connectionId;
+    }
+
+    /**
+     * Test an existing connection status by connectionId.
+     *
+     * @param connectionId the existing connection Id
+     * @return Boolean
+     */
+    public Boolean testExistingConnection(Integer connectionId)
+        throws BadRequestException, BadGatewayException {
+        if (connectionId == null) {
+            throw new BadRequestException(
+                "Connection " + ErrorConstants.ERROR_MISSING_PATH_VARIABLE.getErrorMessage(),
+                HttpStatus.BAD_REQUEST);
+        }
+        boolean testConnectionStatus = false;
+        try {
+            testConnectionStatus = restDriver.testExistingConnection(connectionId);
+        } catch (DgDrvException e) {
+            logger.error("Test connection failed: " + e);
```

Comment:
**Reporter**: PMD    
    **Rule**: GuardLogStatement
    **Severity**: ERROR
    **File**: /work/src/main/java/com/dataguise/saas/service/ConnectionService.java L303
    
    Logger calls should be surrounded by log level guards.&#10;&#10;&#10;Best Practices https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#guardlogstatement
    
*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-273687543>*